src/Utilities/Clock.h

```diff
@@ -45,40 +45,5 @@ class FakeChronoClock
   static duration fake_chrono_clock_increment;
 };
 
-
-/** functor for fake clock
- * calling FakeCPUClock()() returns the clock value
- */
-class FakeCPUClock
-{
-public:
-  static double fake_cpu_clock_value;
-  static double fake_cpu_clock_increment;
-
-  double operator()()
-  {
-    fake_cpu_clock_value += fake_cpu_clock_increment;
-    return fake_cpu_clock_value;
-  }
-};
-
-/** functor for high precision clock
- * calling CPUClock()() returns the clock value
- */
-class CPUClock
-{
-public:
-  double operator()()
-  {
-#ifdef _OPENMP
```

Comment:
Could you refine the header inclusion?
No need of "Concurrency/OpenMP.h"
also change stddef.h to cstdef.
No need of "sys/time.h"

Can we also have alias std::chrono::system_clock to ChronoClock? Should help readability when comparing to FakeChronoClock.